blocks/form/trigger.js

```diff
@@ -0,0 +1,72 @@
+import { openModal } from '../modal/modal.js';
+
+let stopTrigger = false;
+let timer;
+
+export function isFormSubmitted() {
+  return sessionStorage.getItem('formSubmitted');
+}
+
+async function triggerHandler(config) {
+  const { path, value } = config;
+  if (!isFormSubmitted()) {
+    if (!sessionStorage.getItem(path)) { // prevent trigger modal from opening if already closed
+      timer = setTimeout(async () => {
+        if (!stopTrigger) { // stop triggering when another modal is open
+          openModal(config.path, config);
```

Comment:
lets make it independent of modal and expose a call which will decided action (modal) and it will also solve cycle dependency issue